File: pypp_cli/test_dir/python/src/sets/operations.py
def set_operations_fn():
    print("SET OPERATIONS RESULTS:")
    a: set[int] = {1, 2, 3}
    if 1 in a:
        print("1 in a")
    if 4 not in a:
        print("4 not in a")
    print(len(a))
    print(min(a))
    print(max(a))

    # TODO later: implement these other set operations
    # print(a | b)
    # print(a & b)
    # print(a - b)
    # print(a ^ b)

File: pypp_cli/test_dir/cpp/src/sets/operations.cpp
#include "src/sets/operations.h"
#include "py_set.h"
#include "py_str.h"
#include "pypp_util/print.h"

namespace me {
void set_operations_fn() {
    pypp::print(pypp::PyStr("SET OPERATIONS RESULTS:"));
    pypp::PySet<int> a({1, 2, 3});
    if (a.contains(1)) {
        pypp::print(pypp::PyStr("1 in a"));
    }
    if (!a.contains(4)) {
        pypp::print(pypp::PyStr("4 not in a"));
    }
    pypp::print(a.len());
    pypp::print(a.min());
    pypp::print(a.max());
}

} // namespace me